packages/contracts/evm-contracts/contracts/orderbook/OrderbookDex.sol

```diff
@@ -120,8 +176,18 @@ contract OrderbookDex is IOrderbookDex, ERC1155Holder, ReentrancyGuard {
                 assetsToBuy,
                 bytes("")
             );
-            order.seller.sendValue(assetsToBuy * order.pricePerAsset);
-            emit OrderFilled(order.seller, orderId, msg.sender, assetsToBuy, order.pricePerAsset);
+            order.seller.sendValue(purchaseCost - makerFee);
+            emit OrderFilled(
+                asset,
+                order.assetId,
+                orderId,
+                order.seller,
+                msg.sender,
+                assetsToBuy,
+                order.pricePerAsset,
+                makerFee,
+                takerFee
+            );
             if (remainingEth == 0) {
```

Comment:
What about if the value is something really close to zero, but not exactly zero because of floating point precision?

Reply:
It will continue iterating through orders, and then at the end if the minimum received asset (which has slippage subtraction applied) is satisfied, fill is successful and any remaining eth is returned to buyer